DemoQA/Tests/elements_webtables.js

```diff
@@ -96,16 +99,20 @@ test(`Edit an existing user`, async t => {
 });
 
 
-test(`Search table for firstName`, async t => {
+test.only('Search table for firstName', async t => { 
     await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
     
-    const firstNameUser = new UserInfoRow(0);
-    const userName = "Cierra"; // should it be more general? so it catches any first Name property from table?
+    const existingUsersDetails = await webTablesPage.fetchValidUserDetails();
+    const randomIndex = Math.floor(Math.random() * existingUsersDetails.length);
+    const randomUser = existingUsersDetails[randomIndex];
+
+    await webTablesPage.enterSearchText(randomUser.firstName);
     
-    await webTablesPage.enterSearchText(userName);
+    const searchedUserRows = await webTablesPage.fetchValidUserDetails();
     
-    const searchedUserName = await firstNameUser.getUserDetails();
+    await t.expect(searchedUserRows.length).eql(1);
     
-    await t.expect(searchedUserName.firstName.trim()).eql(userName);
-  });
-  
\ No newline at end of file
+    const searchedUserFirstName = searchedUserRows.find(webTablesPage.firstNameValue);
```

Comment:
I'm not sure how this is not failing. Anyway, on row 113 if you already have an expect that you should only find 1 user. Why not expect searchUserRows[0].firstName to be the firstName that you searched for?